src/liballoc/sync.rs

```diff
@@ -1277,7 +1275,7 @@ impl<T: ?Sized> Arc<T> {
     #[inline]
     #[unstable(feature = "get_mut_unchecked", issue = "63292")]
     pub unsafe fn get_mut_unchecked(this: &mut Self) -> &mut T {
-        &mut this.ptr.as_mut().data
+        &mut (*this.ptr.as_ptr()).data
```

Comment:
Maybe add a comment saying that we carefully do *not* create a reference covering the "count" fields.

Reply:
👍